internal/v1/handler_blueprints.go

```diff
@@ -199,3 +201,68 @@ func (h *Handlers) GetBlueprints(ctx echo.Context, params GetBlueprintsParams) e
 		Data: data,
 	})
 }
+
+func (h *Handlers) GetBlueprintComposes(ctx echo.Context, blueprintId openapi_types.UUID, params GetBlueprintComposesParams) error {
+	idHeader, err := getIdentityHeader(ctx)
+	if err != nil {
+		return err
+	}
+
+	limit := 100
+	if params.Limit != nil {
+		if *params.Limit > 0 {
+			limit = *params.Limit
+		}
+	}
+
+	offset := 0
+	if params.Offset != nil {
+		offset = *params.Offset
+	}
+	ignoreImageTypeStrings := convertIgnoreImageTypeToSlice(params.IgnoreImageTypes)
+
+	since := time.Hour * 24 * 14
+
+	composes, err := h.server.db.GetBlueprintComposes(idHeader.Identity.OrgID, blueprintId, params.BlueprintVersion, since, limit, offset, ignoreImageTypeStrings)
+	if err != nil {
+		return err
+	}
+	count, err := h.server.db.CountBlueprintComposesSince(idHeader.Identity.OrgID, blueprintId, params.BlueprintVersion, since, ignoreImageTypeStrings)
+	if err != nil {
+		return err
+	}
+
+	data := make([]ComposesResponseItem, 0, len(composes))
+	for _, c := range composes {
+		bId := c.BlueprintId
+		version := c.BlueprintVersion
+		clientId := (ClientId)(*c.ClientId)
```

Comment:
I've learned today. If I don't do this, the values for all entries are overriden for pointer values, apparently the memory in the cycle is reused and the struct values are overriden in the same memory.
Interesting.

Reply:
I already did hit this several times myself, yep. A nasty one.